x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/persistence/InferenceIndexConstants.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.core.ml.inference.persistence;
+
+/**
+ * Class containing the index constants so that the index version, name, and prefix are available to a wider audience.
+ */
+public final class InferenceIndexConstants {
+
+    public static final String INDEX_VERSION = "000001";
```

Comment:
this looks like a rollover pattern, I think version should be non-fixed size digits (-> single digit in this case) and the rollover pattern should be an additional suffix if rollover is wanted.
Are you planning to use rollover for this in future to delete old models? It seems that configs and models are stored in this index, that would rule out rollover.

Reply:
@hendrikmuhs This is for index versioning. Simply using `1` won't work.

Reply:
you mean because of sorting on retrieval and there limitations? ok, 1 digit limits to 9 versions. But almost 1 million versions seems a little bit to much.

Reply:
I think a single extension for both potential future rollover and mappings version is OK.  The rollover API is designed with periodic changes to index templates in mind.  And then since rollover uses six digits it's best to use six digits.  If we use fewer and then decide to do rollover then it forces us to have two suffices.

Reply:
for support reasons I would prefer 2 suffixes, but you have a point.

It's a good hint, I am currently using 1 digit in transforms, which limits versioning to 9 versions. I will upgrade to use more digits, 6 sounds still to much as I rule out rollover for this index.